src/core/parser.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <initializer_list>
+#include <nlohmann/json.hpp>
+#include "parse_feeder.hpp"
+#include "parse_worker.hpp"
+#include "comments_parser.hpp"
+#include "ignore_parser.hpp"
+
+namespace docgen {
+namespace core {
+
+class Parser : private ParseWorker
+{
+	public:
+		Parser()
+			: ParseWorker {
+				CommentParser(Symbol::BEGIN_LINE_COMMENT, Symbol::NEWLINE),
+				CommentParser(Symbol::BEGIN_BLOCK_COMMENT, Symbol::END_BLOCK_COMMENT, {
+					IgnoreParser(Symbol::NEWLINE, Symbol::STAR)
+				})
+			}
+		{}
+
+		void process(const std::vector<token_t>& tokens);
+
+		const nlohmann::json& parsed() const { return parsed_; }
+
+	private:
+		nlohmann::json parsed_;
+};
+
+inline void Parser::process(const std::vector<token_t>& tokens)
+{
+	ParseFeeder::reset();
+
+	for (const token_t& token : tokens) {
+		proc(token);
```

Comment:
When calling base class member functions, I prefer to use `this->proc`. It clarifies that `proc` is a member function but once user sees it is not a member function listed in `Parser` they can easily deduce that it must come from the base class.

Reply:
True, I agree.